File: Leetcode 0532 K-diff Pairs in an Array [medium] [array]/K-diff Pairs in an Array [hmap].java
class Solution {
    public int findPairs(int[] nums, int k) {
        if (k < 0) {
            return 0;
        }
        
        int res = 0;
        
        Map<Integer, Integer> map = new HashMap<>();
        for (int n : nums) {
            map.put(n, map.getOrDefault(n, 0) + 1);
        }
        
        for (int n : map.keySet()) {
            if (k == 0) {
                if (map.get(n) >= 2) {
                    res++;
                }
            } 
            
            if (k > 0) {
                if (map.containsKey(n+k)) {
                    res++;
                }
            }
        }
        
        return res;
    }
}

File: Leetcode 0532 K-diff Pairs in an Array [medium] [array]/K-diff Pairs in an Array [hmap].js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findPairs = function(nums, k) {
    if (k < 0) {
        return 0;
    }
    
    let res = 0;
    let map = new Map();
    for (let n of nums) {
        map.set(n, map.get(n) + 1 || 1);
    }
    
    for (let n of map.keys()) {
        if (k == 0) {
            if (map.get(n) >= 2) {
                res++;
            }
        }
        
        if (k > 0) {
            if (map.has(n + k)) {
                res++;
            }
        }
    }
    
    return res;
};